tests/tracer/test_utils.py

```diff
@@ -260,6 +261,19 @@ def test_func_name(self):
 def test_infer_arg_value_hit(args, kwargs, pos, kw, expected):
     assert get_argument_value(args, kwargs, pos, kw) == expected
 
+@pytest.mark.parametrize(
+    "args,kwargs,pos,kw,new_value",
+    [
+        ((), {"foo": 42, "bar": "snafu"}, 0, "foo", 4242),
+        ((), {"foo": 42, "bar": "snafu"}, 1, "bar", "new_snafu"),
+        ((42,), {"bar": "snafu"}, 0, "foo", 4422),
+        ((42, "snafu"), {}, 1, "bar", "new_snafu"),
+    ],
+)
+def test_set_argument_value(args, kwargs, pos, kw, new_value):
+    new_args, new_kwargs = set_argument_value(args, kwargs, pos, kw, new_value)
+    assert get_argument_value(new_args, new_kwargs, pos, kw) == new_value
```

Comment:
hmm... I wonder if we should assert the actual provided `new_args` and `new_kwargs` instead of using `get_argument_value`. If there is a bug introduced in `get_argument_value` it could change the validity of this test. Think, if we accidentally update kwargs instead of args, but then `get_argument_value` pulls the new value from kwargs so the test passes.

We do have tests for `get_argument_value` so maybe I am being overly paranoid? wdyt?

Reply:
I think we have sufficient tests for `get_argument_value` so it should be fine. 

 I think it would be overkill to reimplement the functionality of `get_argument_value()` just for these two tests.

Reply:
It isn't re-implementing. given certain inputs, the outputs are static/known.

e.g. given `([], {"foo": 42, "bar": "snafu"}, 0, "foo", 4242)`, the expected result is `([], {"foo": 42, "bar": "snafu"})`